hw/dv/tools/dvsim/common_sim_cfg.hjson

```diff
@@ -84,30 +84,25 @@
   ]
 
   // Build modes are collection of build_opts and run_opts
-  // A test can enable a specific build mode by setting 'use_build_mode' key
-  build_modes: [
-    {
-      name: foo
-      build_opts: ["+define+bx",
-                   "+define+by",
-                   "+define+bz"]
-      run_opts: ["+rx=1",
-                 "+ry=2",
-                 "+rz=3"]
-    }
-    {
-      name: bar
-      build_opts: ["+define+bbaru1",
-                   "+define+bbaru2",
-                   "+define+bbaru3"]
-      run_opts: ["+rbar1u=1",
-                 "+rbar2u=2",
-                 "+rbar3u=3"]
-    }
-    {
-      name: cover_reg_top
-    }
```

Comment:
This needs to be retained.

Reply:
Oh, really? I grepped through and every sim config that uses `cover_reg_top` also defines it. What does retaining it here do?

Reply:
It is used to differentiate the builds for the common automated tests (CSR, MEM etc). It kicks in only when coverage is enabled (they use a diffferent coverage hierarchy file which is supplied during build). 

```
grep -nir cover_reg_top hw/*
hw/dv/tools/dvsim/common_sim_cfg.hjson:108:      name: cover_reg_top
hw/dv/tools/dvsim/common_sim_cfg.hjson:154:  cover_reg_top_vcs_cov_cfg_file: "-cm_hier {tool_srcs_dir}/cover_reg_top.cfg"
hw/dv/tools/dvsim/common_sim_cfg.hjson:162:  // cover_reg_top_xcelium_cov_cfg_file: "-covfile {tool_srcs_dir}/cover_reg_top.ccf"
hw/dv/tools/dvsim/tests/csr_tests.hjson:7:      name: cover_reg_top
hw/dv/tools/dvsim/tests/csr_tests.hjson:22:      build_mode: "cover_reg_top"
hw/dv/tools/dvsim/tests/csr_tests.hjson:29:      build_mode: "cover_reg_top"
hw/dv/tools/dvsim/tests/csr_tests.hjson:36:      build_mode: "cover_reg_top"
hw/dv/tools/dvsim/tests/csr_tests.hjson:43:      build_mode: "cover_reg_top"
hw/dv/tools/dvsim/tests/csr_tests.hjson:50:      build_mode: "cover_reg_top"
hw/dv/tools/dvsim/tests/csr_tests.hjson:57:      build_mode: "cover_reg_top"
hw/dv/tools/dvsim/tests/tl_access_tests.hjson:7:      name: cover_reg_top
hw/dv/tools/dvsim/tests/tl_access_tests.hjson:14:      build_mode: "cover_reg_top"
hw/dv/tools/dvsim/tests/mem_tests.hjson:7:      name: cover_reg_top
hw/dv/tools/dvsim/tests/mem_tests.hjson:22:      build_mode: "cover_reg_top"
hw/dv/tools/dvsim/tests/mem_tests.hjson:29:      build_mode: "cover_reg_top"
hw/dv/tools/dvsim/tests/intr_test.hjson:7:      name: cover_reg_top
hw/dv/tools/dvsim/tests/intr_test.hjson:14:      build_mode: "cover_reg_top"
hw/dv/tools/dvsim/tests/shadow_reg_errors_tests.hjson:7:      name: cover_reg_top
hw/dv/tools/dvsim/tests/shadow_reg_errors_tests.hjson:23:      build_mode: "cover_reg_top"
hw/top_earlgrey/dv/chip_sim_cfg.hjson:47:    // and the 'cover_reg_top' (used by common tests) builds. We override the
hw/top_earlgrey/dv/chip_sim_cfg.hjson:59:    // Used by 'cover_reg_top' only builds - we only cover the *_reg_top of
hw/top_earlgrey/dv/chip_sim_cfg.hjson:63:      name: cover_reg_top_vcs_cov_cfg_file
hw/top_earlgrey/dv/chip_sim_cfg.hjson:64:      value: "-cm_hier {tool_srcs_dir}/chip_cover_reg_top.cfg"
```

Reply:
Ok, but my point is that each of those files *defines* `cover_reg_top` as well as using it. Maybe I've misunderstood something: can you point me at a command that breaks if we remove it from `common_sim_cfg.hjson`?

Reply:
Hmm, you're right - tests are passing with this removed... I vaguely remember needing this for something but I cannot remember why!